go-controller/pkg/node/gateway_udn.go

```diff
@@ -757,6 +752,16 @@ func (udng *UserDefinedNetworkGateway) getDefaultRoute(isNetworkAdvertised bool)
 	return retVal, nil
 }
 
+func (udng *UserDefinedNetworkGateway) getDefaultRouteWithAdvertisedCheck() ([]netlink.Route, error) {
```

Comment:
lol 
I liked my method name suggestion better because it was more clear on what it does and it is an un-exported method anyway.
I guess you didn't. 
I will learn to accept this is subjective so let's go with your option.